packages/opentelemetry-exporter/src/index.ts

```diff
@@ -0,0 +1,211 @@
+    if (endpointUrl) {
+      process.env[instanaEndpointUrlEnvVar] = endpointUrl;
+    }
+
+    if (agentKey) {
+      process.env[instanaAgentKeyEnvVar] = agentKey;
+    }
+
+    instanaEnvironment.validate();
+    instanaBackendConnector.init();
+  }
+
+  export(spans: ReadableSpan[], resultCallback: (result: ExportResult) => void): void {
+    const validSpans = spans.map(this._transform.bind(this));
+    this._sendSpans(validSpans, resultCallback);
+  }
+
+  shutdown(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  private _sendSpans(spans: any[], resultCallback: (result: ExportResult) => void) {
+    instanaBackendConnector.sendSpans(spans,  (err: any, _responseBody?: string) => {
+      let result: ExportResult = { code: ExportResultCode.SUCCESS };
+      if (err) {
+        console.log('error', err);
+        result = { code: ExportResultCode.FAILED, error: err };
+      }
+      resultCallback(result);
+    });
+  }
+
+  private _transform(span: ReadableSpan): any {
+
+    let result: { [key: string]: any } = {
+      n: 'otel',
+      f: {
+        e: process.pid
+      },
+      data: {}
+    };
+
+    const traceId = this._buildTraceOrSpanId(span.spanContext().traceId, true);
+
+    if (traceId.length === 32) {
+      result.t = traceId.substr(16);
+    } else {
+      result.t = traceId;
+    }
+
+    const hasParent = typeof span.parentSpanId === 'string' && span.parentSpanId.length > 0;
+
+    if (hasParent) {
+      result.p = this._buildTraceOrSpanId(span.parentSpanId as string);
+    }
+    
+    result.s = this._buildTraceOrSpanId(span.spanContext().spanId);
+    result.ts = span.startTime[0] * 1000;
+    result.d = (span.endTime[0] - span.startTime[0]) * 1000;
+
+    let isEntrySpan = false;
+
+    switch (span.kind) {
+      case SPAN_KIND_SERVER:
+        isEntrySpan = true;
+        result.data.kind = SERVER_KIND;
+        break;
+      case SPAN_KIND_CLIENT:
+        result.data.kind = CLIENT_KIND;
+        break;
+      case SPAN_KIND_PRODUCER:
+        result.data.kind = PRODUCER_KIND;
+        break;
+      case SPAN_KIND_CONSUMER:
+        isEntrySpan = true;
+        result.data.kind = CONSUMER_KIND;
+        break;
+      case SPAN_KIND_INTERNAL:
+        result.data.kind = INTERNAL_KIND;
+        break;
+      default:
+        break;
+      }
+
+    if (hasParent && isEntrySpan) {
+      // If an OTel entry span continues an ongoing trace (which is equivalent to the original span having a parent), it
+      // always uses the IDs from the traceparent header, thus we mark the span with span.tp accordingly.
+      result.tp = true;
+    }
+
+    result.data.service = span.resource && span.resource.attributes && span.resource.attributes['service.name'];
+    result.data.operation = span.name;
+
+    if (span.spanContext().traceState) {
+      // TODO: should we stringify and parse this guy?
+      result.data.trace_state = span.spanContext().traceState;
```

Comment:
I see that this is in the agent code, too, but I wonder why we even capture the tracestate header in full? We should discuss that with Miel and the others I think. 🤔